### PR TITLE
DTSPO-14238 Initial module

### PR DESCRIPTION
**Pull request by timja**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPO-14238

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,32 +32,39 @@ module "todo_resource_name" {
...
 
 ## Inputs
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| <a name="input_common_tags"></a> [common\_tags](#input\_common\_tags) | Common tag to be applied to resources. | `map(string)` | n/a | yes |
-| <a name="input_component"></a> [component](#input\_component) | https://hmcts.github.io/glossary/#component | `string` | n/a | yes |
-| <a name="input_env"></a> [env](#input\_env) | Environment value. | `string` | n/a | yes |
-| <a name="input_existing_resource_group_name"></a> [existing\_resource\_group\_name](#input\_existing\_resource\_group\_name) | Name of existing resource group to deploy resources into | `string` | `null` | no |
+| <a name="input_application_type"></a> [application\_type](#input\_application\_type) | Specifies the type of Application Insights to create. Valid values are `java` for Java web, `Node.JS` for Node.js, `other` for General, and `web` for ASP.NET | `string` | `"web"` | no |
```
[the remark this answers is not in the document]
Comment (timja) [reply README.md:51]: it's defaulted to web as that's what all of our instances use

[the remark this answers is not in the document]
Comment (thomast1906) [reply README.md:51]: ah, reading further down this PR - we use `web` as default

File: inputs-required.tf [hunk 2]
```diff
@@ -1,23 +1,14 @@
 variable "env" {
-  description = "Environment value."
+  description = "Environment value"
```
Comment (thomast1906) [inline inputs-required.tf:2]: can we put these changes into the base terraform template?

Comment (timja) [reply inputs-required.tf:2]: https://github.com/hmcts/terraform-module-template/commit/d42e1b740d31064495c533be4a4daec8f44128af

File: inputs-optional.tf [hunk 3]
```diff
@@ -11,6 +10,22 @@ variable "location" {
...
+  default     = null
+  type        = string
+}
+
+variable "application_type" {
+  default     = "web" # most of our instances have historically used web, even if other might be more appropriate
+  description = "Specifies the type of Application Insights to create. Valid values are `java` for Java web, `Node.JS` for Node.js, `other` for General, and `web` for ASP.NET"
+}
+
+variable "sampling_percentage" {
```
[the remark this answers is not in the document]
Comment (timja) [reply inputs-optional.tf:23]: I don't think we want to lower sampling by default, probably depends on the app.

`daily_data_cap_in_gb` we could lower, should we do 50 by default?

[the remark this answers is not in the document]
Comment (thomast1906) [reply inputs-optional.tf:23]: sure, yeah lets go with 50 (i'd probably go lower, but teams might just self adjust)

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**